app/questionnaire/questionnaire_schema.py

```diff
@@ -215,20 +216,49 @@ def get_post_submission(self) -> ImmutableDict:
         schema: ImmutableDict = self.json.get("post_submission", ImmutableDict({}))
         return schema
 
+    def _is_list_name_in_rule(
+        self, rules: Union[dict, Sequence], list_name: str
+    ) -> bool:
+
+        if isinstance(rules, dict) and any(
+            operator in rules for operator in OPERATION_MAPPING
+        ):
+            rules = self.get_operands(rules)
+
+        for rule in rules:
+            if not isinstance(rule, dict):
+                continue
+
+                # old rules
+            if "list" in rule:
+                return rule.get("list") == list_name
+
+                # New rules
+            if "source" in rule:
+                return (
+                    rule.get("source") == "list" and rule.get("identifier") == list_name
+                )
+
+                # Nested rules
```

Comment:
Minor, but the indentation is off.
```suggestion
            # Old rules
            if "list" in rule:
                return rule.get("list") == list_name

            # New rules
            if "source" in rule:
                return (
                    rule.get("source") == "list" and rule.get("identifier") == list_name
                )

            # Nested rules
```